src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -464,23 +473,17 @@ public bool ApplyNullableTransforms(ImmutableArray<bool> transforms, bool useNon
 
             if (!result.IsValueType)
             {
-                if (isNullable)
+                if (isAnnotated)
```

Comment:
It may make sense to replace `AsNullableReferenceType` and `AsNotNullableReferenceType` with a single with'er for `IsAnnotated`.

Reply:
We could, although these methods are specific to reference types, but a `WithIsAnnotated` would probably need to handle value types as well. Will leave as is for now.

---
In reply to: [203863171](https://github.com/dotnet/roslyn/pull/28687#discussion_r203863171) [](ancestors = 203863171)

Reply:
I don't think we would need to handle value types differently. This method should apply `IsAnnotated` and `NonNullTypes` information from metadata almost without questions. We just want to check that the combination is valid.

I'm ok to leave for now. I'll take a note to revisit as part of NonNullTypes/PE work.